apps/web/src/pages/integrations/UpdateProviderPage.tsx

```diff
@@ -0,0 +1,339 @@
+import { useFetchEnvironments } from '../../hooks/useFetchEnvironments';
+import { ProviderImage } from './components/multi-provider/SelectProviderSidebar';
+import { When } from '../../components/utils/When';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { updateIntegration } from '../../api/integration';
+import { QueryKeys } from '../../api/query.keys';
+import { errorMessage, successMessage } from '../../utils/notifications';
+
+interface IProviderForm {
+  name: string;
+  credentials: ICredentialsDto;
+  active: boolean;
+  identifier: string;
+}
+
+export function UpdateProviderPage() {
+  const { environments, isLoading: areEnvironmentsLoading } = useFetchEnvironments();
+  const [selectedProvider, setSelectedProvider] = useState<IIntegratedProvider | null>(null);
+  const { providers, refetch, isLoading } = useProviders();
+  const { integrationId } = useParams();
+  const navigate = useNavigate();
+  const queryClient = useQueryClient();
+
+  const { mutateAsync: updateIntegrationApi, isLoading: isLoadingUpdate } = useMutation<
+    { res: string },
+    { error: string; message: string; statusCode: number },
+    {
+      id: string;
+      data: IUpdateIntegrationBodyDto;
+    }
+  >(({ id, data }) => updateIntegration(id, data));
+
+  const {
+    control,
+    handleSubmit,
+    reset,
+    watch,
+    setValue,
+    formState: { errors, isDirty },
+  } = useForm<IProviderForm>({
+    shouldUseNativeValidation: false,
+    defaultValues: {
+      name: '',
+      credentials: {},
+      active: false,
+      identifier: '',
+    },
+  });
+
+  const credentials = watch('credentials');
+
+  const haveAllCredentials = useMemo(() => {
+    if (selectedProvider === null) {
+      return false;
+    }
+    const missingCredentials = selectedProvider.credentials
+      .filter((credential) => credential.required)
+      .filter((credential) => {
+        const value = credentials[credential.key];
+
+        return !value;
+      });
+
+    return missingCredentials.length === 0;
+  }, [selectedProvider, credentials]);
+
+  const identifierClipboard = useClipboard({ timeout: 1000 });
+
+  async function onUpdateIntegration(data) {
+    try {
+      await updateIntegrationApi({
+        id: selectedProvider?.integrationId ?? '',
+        data: { ...data },
+      });
+
+      await queryClient.refetchQueries({
+        predicate: ({ queryKey }) => queryKey.includes(QueryKeys.integrationsList),
+      });
+
+      successMessage('Instance configuration updated');
+    } catch (e: any) {
+      errorMessage(e.message || 'Unexpected error');
+    }
+  }
+
+  useEffect(() => {
+    if (selectedProvider && !selectedProvider?.identifier) {
+      const newIdentifier = slugify(selectedProvider?.displayName, {
+        lower: true,
+        strict: true,
+      });
+
+      setValue('identifier', newIdentifier);
+    }
+  }, [selectedProvider]);
```

Comment:
It is just for the case of an existing integrations. I believe after migration it will no longer be needed, but still good to have to cover it.